### PR TITLE
Creating server-side directory

### PR DESCRIPTION
**Pull request by jking31cs**
Added a README and basic build files with dependencies we will need.

### CODE REVIEW

**Comment by jking31cs** [comment]
All this does is setup the gradle build folder with the Amazon libraries we need, and I wrote up a readme on server side code dev.  Let me know if it makes sense or not.

File: server/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+We'll handle dependencies between classes using Google Guice: https://github.com/google/guice
+* Each feature package will contain a module
+* Each request handler will create an injector to use in its constructor.
+  * Note that the injector should include everything required for that handler class.
+
+The general structure should be as follows:
+* All Echo specific Code belongs in a Speechlet class.
+  * For example, `MedSpeechlet.java` would be a Speechlet for med adding/removing
+  * Follow https://developer.amazon.com/appsandservices/solutions/alexa/alexa-skills-kit/getting-started-guide for conventions.
+* All Rest endpoints should beling in a WebService class.
```
Comment (mckoon) [inline server/README.md]: beling -> belong

File: server/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+In `build/distributions`, a file called `server.zip` will be created that contains all dependency jars as well as compiled classes.
+This is what is added to the lambdas
+
+## Code Conventions
+All classes will be contained in a package associated with the feature it belongs to.
+For example `Med.java` would belong in the `info.doseamigos.med` package.
+
+The code will be written in Java 8, so that it's compatible with Amazon AWS Lambdas.
+
+We'll handle dependencies between classes using Google Guice: https://github.com/google/guice
```
Comment (jking31cs) [inline server/README.md:15]: I looked up other examples of people using Spring with lambdas, and it looks like a giant hassle either way.  I find guide to be more lightweight and easier to understand myself.  What sucks is that we'll probably have to create injectors in each handler that we specify to lambda.   Not the end of the world, just not great.

File: server/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* Each feature package will contain a module
+* Each request handler will create an injector to use in its constructor.
+  * Note that the injector should include everything required for that handler class.
+
+The general structure should be as follows:
+* All Echo specific Code belongs in a Speechlet class.
+  * For example, `MedSpeechlet.java` would be a Speechlet for med adding/removing
+  * Follow https://developer.amazon.com/appsandservices/solutions/alexa/alexa-skills-kit/getting-started-guide for conventions.
+* All Rest endpoints should beling in a WebService class.
+  * Can have mutliple Handler methods for each http method
```
Comment (mckoon) [inline server/README.md]: mutliple -> multiple

Comment (jking31cs) [inline server/README.md]: My thinking here is that we can specify multiple handlers in the same class, as that seems possible with Amazon Lambda

**Comment by jking31cs** [comment]
Fixed spelling errors because typing is hard...

File: server/build.gradle [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+//Dependencies for the project
+dependencies {
+    compile 'org.slf4j:slf4j-api:1.7.12'
+    compile 'com.amazonaws:aws-lambda-java-core:1.1.0'
+    compile 'com.amazonaws:aws-lambda-java-events:1.1.0'
+    compile 'mysql:mysql-connector-java:6.0.2'
+    compile 'com.google.inject:guice:4.1.0'
+
+    testCompile 'org.testng:testng:6.9.10'
```
Comment (mckoon) [inline server/build.gradle:18]: You add TestNG dependency, but where are you telling gradle to run it during the test phase?

https://docs.gradle.org/current/dsl/org.gradle.api.tasks.testing.Test.html

Comment (jking31cs) [inline server/build.gradle:18]: Fixed that, whoops.

**Comment by jking31cs** [comment]
Made the test task in gradle use testNG

File: server/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Dose Amigos Server Side Code
```
Comment (mckoon) [inline server/README.md:1]: It doesn't really hurt anything, but the server folder is unnecessary -- all of this repo is server-side code.

The ionic2 code will be in a separate dose-amigos-ionic repo.

**Comment by mckoon** [comment]
+1

**Comment by geoffreygraddy** [comment]
+1

**Comment by jking31cs** [comment]
Got majority approval, merging.

**Pull request merged**